data/text_sampling/extend_tabular.py

```diff
@@ -0,0 +1,254 @@
+import glob
+import multiprocessing as mp
+import os
+import random
+import time
+from functools import partial
+
+import deepsmiles
+import pandas as pd
+import pubchempy as pcp
+import requests
+import selfies
+from rdkit import Chem
+
+# tucan needs very likely python 3.10
+# from tucan.canonicalization import canonicalize_molecule
+# from tucan.io import graph_from_molfile_text
+# from tucan.serialization import serialize_molecule
+from utils import load_yaml
+
+
+def augment_smiles(smiles: str, int_aug: int = 50, deduplicate: bool = True) -> str:
+    """
+    Takes a SMILES (not necessarily canonical) and returns `int_aug` random variations of this SMILES.
+    """
+
+    mol = Chem.MolFromSmiles(smiles)
+
+    if mol is None:
+        return None
+    else:
+        if int_aug > 0:
+            augmented = [
+                Chem.MolToSmiles(mol, canonical=False, doRandom=True)
+                for _ in range(int_aug)
+            ]
+            if deduplicate:
+                augmented = list(set(augmented))
+            return augmented
+        else:
+            raise ValueError("int_aug must be greater than zero.")
+
+
+def smiles_to_selfies(smiles: str) -> str:
+    """
+    Takes a SMILES and return the selfies encoding.
+    """
+
+    return [selfies.encoder(smiles)]
```

Comment:
```suggestion
    return selfies.encoder(smiles)
```

perhaps I don't get it, but why is this one a list? (here, the type annotation would also be incorrect)

Reply:
huh, good catch, it should be a str and not a list, I removed the brackets again.

Reply:
removed and tested